src/core/friendly_errors/param_validator.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * @for p5
+ * @requires core
+ */
+// import p5 from '../main';
+import dataDoc from '../../../docs/parameterData.json';
+import { z } from 'zod';
+import { fromError } from 'zod-validation-error';
+
+let schemaRegistry = new Map();
+const arrDoc = JSON.parse(JSON.stringify(dataDoc));
+
+const schemaMap = {
+  'Any': z.any(),
+  'Array': z.array(z.any()),
+  'Boolean': z.boolean(),
+  'Function': z.function(),
+  'Integer': z.number().int(),
+  'Number': z.number(),
+  'Number[]': z.array(z.number()),
+  'Object': z.object({}),
+  // Allows string for any regex
+  'RegExp': z.string(),
+  'String': z.string(),
+  'String[]': z.array(z.string())
+};
+
+function generateZodSchemasForFunc(func) {
+  const ichDot = func.lastIndexOf('.');
```

Comment:
It might be helpful to readers to have a comment above this with some examples of the format of `func` that we expect. Based on the code below it looks like global functions might just be e.g. `sin` and class methods might be like `p5.Vector.add`?

Reply:
I got this part from the original validate_params file and I believe that's the case!